pkg/segment/aggregations/evalaggs.go

```diff
@@ -600,6 +600,35 @@ func PerformAggEvalForCardinality(measureAgg *structs.MeasureAggregator, strSet
 	return float64(len(strSet)), nil
 }
 
+func PerformAggEvalForList(measureAgg *structs.MeasureAggregator, strList []string, fieldToValue map[string]utils.CValueEnclosure) ([]string, error) {
+	fields := measureAgg.ValueColRequest.GetFields()
+
+	if len(fields) == 0 {
+		valueStr, err := measureAgg.ValueColRequest.EvaluateToString(fieldToValue)
+		if err != nil {
+			return []string{}, fmt.Errorf("PerformAggEvalForList: Error while evaluating value col request function: %v", err)
+		}
+		strList = append(strList, valueStr)
+	} else {
+		if measureAgg.ValueColRequest.BooleanExpr != nil {
+			boolResult, err := measureAgg.ValueColRequest.BooleanExpr.Evaluate(fieldToValue)
+			if err != nil {
+				return []string{}, fmt.Errorf("PerformAggEvalForList: there are some errors in the eval function that is inside the values function: %v", err)
+			}
+			if boolResult {
+				strList = append(strList, "1")
+			}
+		} else {
+			cellValueStr, err := measureAgg.ValueColRequest.EvaluateToString(fieldToValue)
+			if err != nil {
+				return []string{}, fmt.Errorf("PerformAggEvalForList: Error while evaluating value col request, err: %v", err)
+			}
+			strList = append(strList, cellValueStr)
+		}
```

Comment:
Lines 607-611 and 622-626 look the same to me. Am I missing something? If they are the same, you should rearrange this code so we only have that block once.

Reply:
Yes, they are the same. 
I have optimized the code.